test/controllers/api_users/permissions_controller_test.rb

```diff
@@ -26,6 +26,69 @@ class ApiUsers::PermissionsControllerTest < ActionController::TestCase
       assert_not_authorised
     end
 
+    should "raise an exception if the user doesn't have access to the application" do
+      sign_in create(:admin_user)
+
+      application = create(:application)
+      api_user = create(:api_user)
+
+      assert_raises(ActiveRecord::RecordNotFound) do
+        get :edit, params: { api_user_id: api_user, application_id: application }
+      end
+    end
+
+    should "raise an exception if the application is retired" do
+      sign_in create(:superadmin_user)
+
+      application = create(:application, retired: true)
+      permission = create(:supported_permission, application:)
+      api_user = create(:api_user,
+                        with_signin_permissions_for: [application],
+                        with_permissions: { application => [permission.name] })
```

Comment:
Do we need the non-signin permissions here?

Reply:
We can view the edit page without the permissions, so we can remove the permissions here.

This can be a copy of the access token test below (with retired and access_token) in it